matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/CommonTestHelper.kt

```diff
@@ -65,32 +63,42 @@ import java.util.concurrent.TimeUnit
 class CommonTestHelper internal constructor(context: Context) {
 
     companion object {
-        internal fun runSessionTest(context: Context, autoSignoutOnClose: Boolean = true, block: (CommonTestHelper) -> Unit) {
+
+        @OptIn(ExperimentalCoroutinesApi::class)
+        internal fun runSessionTest(context: Context, autoSignoutOnClose: Boolean = true, block: suspend CoroutineScope.(CommonTestHelper) -> Unit) {
             val testHelper = CommonTestHelper(context)
-            return try {
-                block(testHelper)
-            } finally {
-                if (autoSignoutOnClose) {
-                    testHelper.cleanUpOpenedSessions()
+            return runTest(dispatchTimeoutMs = TestConstants.timeOutMillis) {
+                try {
+                    withContext(Dispatchers.Default) {
```

Comment:
we use the `Default` context to allow `Timeouts` to work correctly, the original `runTest` dispatcher uses a test scheduler which requires the scheduler timings to be manually incremented (which we don't want!)